python/torch_mlir_e2e_test/configs/torchdynamo.py

```diff
@@ -45,12 +49,13 @@ def _refbackend_torchdynamo_backend(fx_graph: torch.fx.GraphModule,
     # Torch-MLIR does not support returning an empty tuple. The reason is
     # that both returning an empty tuple and returning `None` results in MLIR
     # functions that have as a return type `()`. In other words, there is no
-    # way of differentiating between the two. Moreover, since Torch-MLIR treats
-    # inputs as having value semantics, graphs that return nothing are no-ops to
-    # Torch-MLIR.
+    # way of differentiating between the two.
     if _returns_empty_tuple(fx_graph):
         return fx_graph
```

Comment:
I think we should just remove this `if`. Currently it will result in an `assert` error saying that `_refbackend_torchdynamo_backend` did not run, but it would be more accurate to let `torch_mlir.compile` run and have `torch_mlir` create the "no empty returns allowed error"

Reply:
I put this back after our offline discussion because what actually happens is you get ~134 new fails that are all like

```
        torch._dynamo.exc.BackendCompilerFailed: backend='compiler_fn' raised:
        Exception: 
        PyTorch TorchScript module -> torch-mlir Object Graph IR import failed with:
        ### Importer C++ Exception:
        see diagnostics
        ### Importer Diagnostics:
        error: unhandled prim::Constant node: %12 : () = prim::Constant[value=()](
```

But failing on just empty return isn't right either because (possibly) dynamo broke the graph and this is just pathological/empty/useless side of the break (i.e., the next pass through, for the same test, will have useful work and will compile). 

Ultimately I don't know what the answer is here - you have to check whether the substantive part of the test actually ran/got compiled but you can't predict when dynamo will break and you can't know after the fact except in some cases (wherein it decides to percolate up by some means, raising an Exception of otherwise).

Reply:
I see, that makes sense. In that case, I'm fine with living it the way it is

Reply:
I tried this locally, and the reason that more failures are caused is that the importer error makes either torchdynamo or our testing framework trip up, and many of the tests that execute after end up failing with the same error. When I ran things, I got around 20 new fails, but when those tests are run individually, they pass.

What we can do is turn the empty graph check into an assert instead:

```python
assert not _returns_empty_tuple(fx_graph), "encountered graph that does not return anything"
```

When I did this locally, I did not get any new failing tests. I don't think we have to worry anymore about an empty graph followed by a non-empty graph, because `nopython` is supposed to guarantee just one graph per test. If this is not the case, we should open a bug upstream.

Reply:
TEAMWORK